src/main/java/io/odpf/firehose/sink/prometheus/builder/TimeSeriesBuilder.java

```diff
@@ -0,0 +1,189 @@
+        List<Map<String, Object>> metricList = getMetricMessage(message, metricNameProtoIndexMapping);
+        List<Cortex.TimeSeries> timeSeriesList = new ArrayList<>();
+        Long metricTimestamp = getMetricTimestamp(message);
+        for (Map<String, Object> metricName : metricList) {
+            cortexTimeSeriesBuilder.clear();
+            buildMetric(metricName.get(METRIC_NAME).toString(), cortexTimeSeriesBuilder, cortexLabelBuilder);
+            for (Map.Entry<String, Object> entry : labelPair.entrySet()) {
+                buildLabels(entry.getKey(), entry.getValue(), cortexTimeSeriesBuilder, cortexLabelBuilder);
+            }
+            buildSample(metricTimestamp, Double.parseDouble(metricName.get(METRIC_VALUE).toString()), cortexTimeSeriesBuilder, cortexSampleBuilder);
+            timeSeriesList.add(cortexTimeSeriesBuilder.build());
+        }
+        return timeSeriesList;
+    }
+
+    private void buildMetric(String metricName, Cortex.TimeSeries.Builder cortexTimeSeriesBuilder, Cortex.LabelPair.Builder cortexLabelBuilder) {
+        cortexLabelBuilder.clear();
+        Cortex.LabelPair metric = cortexLabelBuilder
+                .setName(PROMETHEUS_LABEL_FOR_METRIC_NAME)
+                .setValue(metricName)
+                .build();
+        cortexTimeSeriesBuilder.addLabels(metric);
+    }
+
+    private void buildLabels(String labelName, Object labelValue, Cortex.TimeSeries.Builder cortexTimeSeriesBuilder, Cortex.LabelPair.Builder cortexLabelBuilder) {
+        cortexLabelBuilder.clear();
+        Cortex.LabelPair label = cortexLabelBuilder
+                .setName(labelName)
+                .setValue(labelValue.toString())
+                .build();
+        cortexTimeSeriesBuilder.addLabels(label);
+    }
+
+    private void buildSample(long timestamp, double value, Cortex.TimeSeries.Builder cortexTimeSeriesBuilder, Cortex.Sample.Builder cortexSampleBuilder) {
+        cortexSampleBuilder.clear();
+        Cortex.Sample sample = cortexSampleBuilder
+                .setTimestampMs(timestamp)
+                .setValue(value)
+                .build();
+        cortexTimeSeriesBuilder.addSamples(sample);
+    }
+
+    /**
+     * Timestamp that will be use as metric timestamp.
+     * the timestamp from message or from current timestamp base on prometheus config
+     *
+     * @param message                           the protobuf message
+     * @return                                  the unix timestamp
+     * @throws InvalidProtocolBufferException   the exception on invalid protobuf
+     */
+    private Long getMetricTimestamp(Message message) throws InvalidProtocolBufferException {
+        return (isEventTimestampEnabled) ? getMillisFromTimestamp(getTimestamp(message, timestampIndex)) : System.currentTimeMillis();
+    }
+
+    private List<Map<String, Object>> getMetricMessage(Message message, Properties protoIndexMapping) {
```

Comment:
let's rename protoIndexMapping to metricNameProtoIndexMapping to help with the intent. WDYT?